dagshub/auth/oauth.py

```diff
@@ -32,20 +19,20 @@ def oauth_flow(
     state = uuid.uuid4()
     middle_man_uuid = uuid.uuid4()
 
-    with concurrent.futures.ThreadPoolExecutor() as executor:
-        future = executor.submit(call_middleman, dagshub_url, middle_man_uuid)
-        link_prompt = f"Go to {dagshub_url}/authorize?state={state}&client_id={client_id}" \
-                      f"&middleman_uuid={middle_man_uuid} to authorize the client."
-        print(link_prompt)
-        res = future.result()
+    link_prompt = f"Go to {dagshub_url}/authorize?state={state}&client_id={client_id}" \
+                  f"&middleman_uuid={middle_man_uuid} to authorize the client."
+    print(link_prompt)
+
+    res = requests.post(
+        f"{dagshub_url}/middleman",
+        data={"uuid": middle_man_uuid},
+    )
```

Comment:
Check what it looks like when you get a timeout

Reply:
<img width="828" alt="image" src="https://user-images.githubusercontent.com/107917894/204862560-8db7472a-df90-458b-baab-c200804d137a.png">
its clear